dot/parachain/dispute/test_helpers.go

```diff
@@ -0,0 +1,113 @@
+package dispute
+
+import (
+	"fmt"
+	"github.com/ChainSafe/gossamer/dot/parachain/dispute/overseer"
+	parachainTypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/dot/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+func dummyCandidateCommitments() parachainTypes.CandidateCommitments {
+	return parachainTypes.CandidateCommitments{
+		UpwardMessages:            nil,
+		HorizontalMessages:        nil,
+		NewValidationCode:         nil,
+		HeadData:                  parachainTypes.HeadData{},
+		ProcessedDownwardMessages: 0,
+		HrmpWatermark:             0,
+	}
+}
+
+func dummyValidationCode() parachainTypes.ValidationCode {
+	return parachainTypes.ValidationCode{1, 2, 3}
+}
+
+func dummyCollator() parachainTypes.CollatorID {
+	return parachainTypes.CollatorID{}
+}
+
+func dummyCollatorSignature() parachainTypes.CollatorSignature {
+	return parachainTypes.CollatorSignature{}
+}
+
+func dummyCandidateDescriptorBadSignature(relayParent common.Hash) parachainTypes.CandidateDescriptor {
+	zeros := common.Hash{}
+	validationCodeHash, err := dummyValidationCode().Hash()
+	if err != nil {
+		panic(err)
+	}
+
+	return parachainTypes.CandidateDescriptor{
+		ParaID:                      0,
+		RelayParent:                 relayParent,
+		Collator:                    dummyCollator(),
+		PersistedValidationDataHash: zeros,
+		PovHash:                     zeros,
+		ErasureRoot:                 zeros,
+		ParaHead:                    zeros,
+		ValidationCodeHash:          validationCodeHash,
+		Signature:                   dummyCollatorSignature(),
+	}
+}
+
+func dummyCandidateReceiptBadSignature(
+	relayParent common.Hash,
+	commitments *common.Hash,
+) (parachainTypes.CandidateReceipt, error) {
+	var (
+		err             error
+		commitmentsHash common.Hash
+	)
+	if commitments == nil {
+		commitmentsHash, err = dummyCandidateCommitments().Hash()
+		if err != nil {
+			return parachainTypes.CandidateReceipt{}, err
+		}
+	} else {
+		commitmentsHash = *commitments
+	}
+
+	return parachainTypes.CandidateReceipt{
+		Descriptor:      dummyCandidateDescriptorBadSignature(relayParent),
+		CommitmentsHash: commitmentsHash,
+	}, nil
+}
+
+func dummyHash() common.Hash {
```

Comment:
I see we do a mix of using this as well as just declaring a common.Hash{}. Is this helper needed?

Reply:
Yeah, prolly not. These tests don't have any checks on the values.